core/src/main/scala/fs2/Chunk.scala

```diff
@@ -174,15 +176,72 @@ object Chunk {
     new Doubles(values, offset, size)
   }
 
+  // justification of the performance of this class: http://stackoverflow.com/a/6823454
+  // note that Class construction checks are nearly free when done at load time
+  // for that reason, do NOT make the extractors into lazy vals or defs!
   private object Respecialization {
     // TODO set by completely random guess. please tune!
-    val Threshold = 1000
+    val Threshold = 0
```

Comment:
We need to tune this parameter.  It basically sets the minimum chunk length, below which we will not attempt to respecialize (as there is a very slight constant-time, chunk-level overhead to doing so).  I'm *guessing* that this value should be fairly low (something on the order of 10), but we should measure.  It almost certainly should not be 0.